Build fcmp without client: cvercmp not found

When building the modpack installer without the client, cvercmp is missing:
```
Scanning dependencies of target freeciv_modpack
[ 97%] Building CXX object tools/fcmp/CMakeFiles/freeciv_modpack.dir/download.cpp.o
/home/dude/src/freeciv21/tools/fcmp/download.cpp:28:10: fatal error: cvercmp.h: No such file or directory                                                                        
 #include "cvercmp.h"
          ^~~~~~~~~~~
compilation terminated.
```
Building fcmp without the client is important for server environments like LT.

--- a/cmake/FreecivDependencies.cmake
+++ b/cmake/FreecivDependencies.cmake
@@ -132,7 +132,9 @@ if (EMSCRIPTEN)
   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -s ERROR_ON_UNDEFINED_SYMBOLS=0")
 endif()
 
-if (FREECIV_BUILD_LIBCLIENT)
+if (FREECIV_BUILD_LIBCLIENT
+    OR FREECIV_ENABLE_FCMP_CLI
+    OR FREECIV_ENABLE_FCMP_QT)
   # Version comparison library (this should really be part of utility/)
   add_subdirectory(dependencies/cvercmp)
 endif()

